include/timer.h

```diff
@@ -37,6 +38,61 @@ constexpr auto PIT_TICK_RATE_KHZ = static_cast<double>(PIT_TICK_RATE) / 1000.0;
 // In this case, we want the period of every 1000 PIT tick events.
 constexpr auto PERIOD_OF_1K_PIT_TICKS = 1000.0 / static_cast<double>(PIT_TICK_RATE);
 
+/* PIT operating modes represented in 3 bits:
+1 to 3       Operating mode :
+                0 0 0 = Mode 0 (interrupt on terminal count)
+                0 0 1 = Mode 1 (hardware re-triggerable one-shot)
+                0 1 0 = Mode 2 (rate generator)
+                0 1 1 = Mode 3 (square wave generator)
+                1 0 0 = Mode 4 (software triggered strobe)
+                1 0 1 = Mode 5 (hardware triggered strobe)
+                1 1 0 = Mode 2 (rate generator, same as 010b)
+                1 1 1 = Mode 3 (square wave generator, same as 011b)
+Refs: http://www.osdever.net/bkerndev/Docs/pit.htm
+      https://wiki.osdev.org/Programmable_Interval_Timer#Operating_Modes
+*/
+enum class PitMode : uint8_t {
+	InterruptOnTC      = 0b0'0'0,
+	OneShot            = 0b0'0'1,
+	RateGenerator      = 0b0'1'0,
+	SquareWave         = 0b0'1'1,
+	SoftwareStrobe     = 0b1'0'0,
+	HardwareStrobe     = 0b1'0'1,
+	RateGeneratorAlias = 0b1'1'0,
+	SquareWaveAlias    = 0b1'1'1,
+	Inactive,
+};
+
+/*  PPI Port B Control Register
+    Bit System   Description
+    ~~~ ~~~~~~   ~~~~~~~~~~~
+    0   XT & PC  Timer 2 gate to speaker output (read+write)
+    1   XT & PC  Speaker data state (read+write)
+    4   XT & PC  Toggles with each read
+    5   XT-only  Toggles with each read
+        PC-only  Mirrors timer 2 gate to speaker output
+    7   XT-only  Clear keyboard buffer
+*/
+
+union PpiPortB {
+	uint8_t data = {0};
+	bit_view<0, 1> timerGateOutput;
+	bit_view<1, 1> speakerOutput;
+	bit_view<4, 1> toggleOnRead;
+	bit_view<5, 1> toggleOnReadXt;
+	bit_view<5, 1> timerGateOutputAlias;
+	bit_view<7, 1> clearKeyboardXt;
+	// virtual view of the timer and speaker fields
+	bit_view<0, 2> timerGateAndSpeakerOutput;
```

Comment:
Why_not_snake_case?

Reply:
Ahh indeed; I actually had my head in `enum` space! 
(these are struct members.. d'oh.  Thanks for catching that :sweat_smile: )